doc/API.md

```diff
@@ -92,6 +92,7 @@ Creates an instance of Libp2p.
 | options.modules | [`Array<object>`](./CONFIGURATION.md#modules) | libp2p [modules](./CONFIGURATION.md#modules) to use |
 | [options.addresses] | `{ listen: Array<string>, announce: Array<string>, noAnnounce: Array<string> }` | Addresses for transport listening and to advertise to the network |
 | [options.config] | `object` | libp2p modules configuration and core configuration |
+| [options.host] | `{agentVersion: string, protocolVersion: string}` | libp2p host options |
```

Comment:
I'm not inclined to allow setting `protocolVersion`. `agentVersion` is fine it has no impact, but any change to  `protocolVersion` warrants a connection closure in the spec (although we dont enforce that here). We _could_ allow it, but we'd need to be very clear about not changing it unless you're intentionally creating an isolated network. ie: "DO NOT change unless you know what you're doing"

Reply:
Yes, that makes sense.

So, `js-ipfs` has supposedly `protocolVersion` assigned to `9000`. But in theory, this is not used anywhere I think.  I am not sure about the reasoning behind that number, but it would need to be changed to libp2p's default value. If so, we can drop the setting of `protocolVersion` unless a new use case appears

Reply:
go-IPFS reports `protocolVersion` as `'ipfs/0.1.0'` so it should probably change to that?

Reply:
Yeah I'm not sure why js-ipfs is trying to use that, but you are correct, it won't change anything we're actually sending over the wire. That should be corrected in js-ipfs. 

Let's leave out being able to configure `protocolVersion` for now. We may add it in the future, but that's a can of worms I'd rather not open right now.